activesupport/lib/active_support/testing/assertions.rb

```diff
@@ -276,6 +279,36 @@ def _assert_nothing_raised_or_warn(assertion, &block)
 
           raise
         end
+
+        def _callable_to_source_string(callable)
+          if defined?(RubyVM::AbstractSyntaxTree) && callable.is_a?(Proc)
+            ast = begin
+              RubyVM::AbstractSyntaxTree.of(callable, keep_script_lines: true)
+            rescue SystemCallError
+              # Failed to get the source somehow
+              return callable
```

Comment:
Also had to harden it a bit, as evidenced by railties test, it can happen that the source is not accessible.